### PR TITLE
Add docs for Cloud Vision OCR

### LINKED ISSUE #1548: Add documentation for Vision OCR feature.
Add documentation for Vision OCR feature.

- Add section in docs.
- Add javadoc to vision configuration properties.

### PR DESCRIPTION
**Pull request by dzou**
Add docs for Cloud Vision OCR. Fixes #1548.

### CODE REVIEW

File: docs/src/main/asciidoc/vision.adoc [hunk 1]
```diff
@@ -28,13 +33,38 @@ dependencies {
 }
 ----
 
-=== Cloud Vision Template
+==== Cloud Vision OCR Dependencies
+
+If you are interested in applying optical character recognition (OCR) on documents for your project, you'll need to add both `spring-cloud-gcp-starter-vision` and `spring-cloud-gcp-starter-storage` to your dependencies.
+The storage starter is necessary because the Cloud Vision API runs OCR on documents stored in Google Cloud Storage buckets.
```
Comment (meltsufin) [inline docs/src/main/asciidoc/vision.adoc]: Isn't it also that it stores the output of the OCR process in the GCS bucket?

File: docs/src/main/asciidoc/vision.adoc [hunk 2]
```diff
@@ -78,6 +108,50 @@ public void processImage() {
...
+
+The `DocumentOcrTemplate` provides the following method for running OCR on a document saved in Google Cloud Storage:
+
+`ListenableFuture<DocumentOcrResultSet> runOcrForDocument(GoogleStorageLocation document, GoogleStorageLocation outputFilePathPrefix)`
+
+The method allows you to specify the location of the document and the output location for where all the JSON output files will be saved in Google Cloud Storage.
+It returns a `ListenableFuture` containing `DocumentOcrResultSet` which contains the OCR content of the document.
+
+NOTE: Running OCR on a document is an operation that can take between several minutes to several hours depending on how large the document is.
+It is recommended to register callbacks to the returned ListenableFuture or ignore it and process the JSON output files at a later point in time.
```
Comment (meltsufin) [inline docs/src/main/asciidoc/vision.adoc]: ... using `readOrcOutputFileSet`.

File: docs/src/main/asciidoc/vision.adoc (hunk 1 above)
Comment (dzou) [reply docs/src/main/asciidoc/vision.adoc]: Done.

File: docs/src/main/asciidoc/vision.adoc (hunk 2 above)
Comment (dzou) [reply docs/src/main/asciidoc/vision.adoc]: Done.

**Comment by dzou** [comment]
> Overall looks good, but one thing we learned from the UXR study is that people love code snippets in the docs. So, maybe you can add some for each of the two methods. Thanks!

Sounds good, done.

**Pull request merged**